src/index.html

```diff
@@ -1,17 +1,184 @@
 <!doctype html>
 <html lang="en">
+
   <head>
     <meta charset="UTF-8">
-    <meta
-      name="viewport"
-      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
+    <meta name="viewport"
+    content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
     >
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>HTML Form</title>
     <link rel="stylesheet" href="./style.css">
   </head>
+
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form action="http://localhost:8080/api" method="post">
+      <fieldset>
+        <legend>Personal information:</legend>
+        <div class="form-field">
+          <label for="surname">Surname:
+            <input 
+              type="text" 
+              name="surname" 
+              autocomplete="off"
+            >
+          </label>
+        </div>
+
+        <div class="form-field">
+          <label for="name">Name:
+            <input 
+              type="text" 
+              name="name" 
+              required 
+              autocomplete="off"
+            >
+          </label>
+        </div>
+      
+        <div class="form-field">
+          <label>
+            How old are You?
+            <input 
+              type="number" 
+              name="age" 
+              value="12" 
+              min="1" 
+              max="100"
+            >
+          </label>
+        </div>
+      
+        <div class="form-field">
+          <label>
+            Full date of birth:
+            <input 
+              type="date" 
+              name="dateOfBirth" 
+              value="mm/dd/yyyy"
+            >
+          </label>
+        </div>
+      
+        <div class="terms">
+          <label>
+            I accept the term of the agreement
+            <input type="checkbox" name="terms">
+          </label>
+        </div>
+      </fieldset>
+
+      <fieldset>
+        <legend>Registration:</legend>
+        <div class="form-field">
+          <label>
+            E-mail:
+            <input 
+              type="email" 
+              name="email" 
+              placeholder="email@example.com" 
+              required
+            >
+          </label>
+        </div>
```

Comment:
Why do you need additional div for each input? Remove them
```suggestion
          <label class="form-field">
            E-mail:
            <input 
              type="email" 
              name="email" 
              placeholder="email@example.com" 
              required
            >
          </label>
```